diff --git a/src/protocols/ColorManagement.cpp b/src/protocols/ColorManagement.cpp
index 5af099287..0831bd7c2 100644
--- a/src/protocols/ColorManagement.cpp
+++ b/src/protocols/ColorManagement.cpp
@@ -169,8 +169,12 @@ wl_client* CColorManagementSurface::client() {
 }
 
 const SImageDescription& CColorManagementSurface::imageDescription() {
-    if (!hasImageDescription())
-        LOGM(WARN, "Reading imageDescription while none set. Returns default or empty values");
+    if (!hasImageDescription()) {
+        if (!m_loggedMissingImageDescription) {
+            LOGM(WARN, "Reading imageDescription while none set. Returns default or empty values");
+            m_loggedMissingImageDescription = true;
+        }
+    }
     return m_imageDescription;
 }
 
 bool CColorManagementSurface::hasImageDescription() {
@@ -179,6 +183,8 @@ bool CColorManagementSurface::hasImageDescription() {
 
 void CColorManagementSurface::setHasImageDescription(bool has) {
     m_hasImageDescription = has;
+    if (has)
+        m_loggedMissingImageDescription = false;
     m_needsNewMetadata    = true;
 }
 
 bool CColorManagementSurface::needsHdrMetadataUpdate() {
     if (!m_needsNewMetadata)
         return false;
     if (m_imageDescription == m_lastImageDescription)
         m_needsNewMetadata = false;
     return m_needsNewMetadata;
 }
 
 bool CColorManagementSurface::isHDR() {
diff --git a/src/protocols/ColorManagement.hpp b/src/protocols/ColorManagement.hpp
index 254c16a4d..5ab1e1061 100644
--- a/src/protocols/ColorManagement.hpp
+++ b/src/protocols/ColorManagement.hpp
@@ -75,6 +75,7 @@ class CColorManagementSurface {
     NColorManagement::SImageDescription m_imageDescription;
     NColorManagement::SImageDescription m_lastImageDescription;
     bool                                m_hasImageDescription = false;
+    bool                                m_loggedMissingImageDescription = false;
     bool                                m_needsNewMetadata    = false;
     hdr_output_metadata                 m_hdrMetadata;
 
